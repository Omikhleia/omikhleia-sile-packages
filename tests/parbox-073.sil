\begin{document}
\neverindent
\nofolios
\script[src=packages/parbox]
% Test case 073 = parbox minimized again, but we need another test for
% long line-broken text and negative spaces
% A: with a negative width explicitly introduced via a kern
% B: negative parindent, ragged right
% C: negative parindent, justified (regular line-breaking)

073A. \parbox[strut=rule, width=37%fw, valign=middle, border=0.5pt, minimize=true]{%
\quad{}\kern[width=-1em]{}Lorem ipsum.}

\script{
SILE.registerCommand("test:fancymargins", function (options, content)
  SILE.settings.set("document.lskip", SILE.length("1em"))
  SILE.settings.set("document.rskip", SILE.nodefactory.hfillglue())
  SILE.settings.set("document.parindent", SILE.length("-1em"))
  SILE.process(content)
end)
}%
073B. \parbox[strut=rule, width=37%fw, valign=middle, border=0.5pt, minimize=true]{%
\test:fancymargins{Lorem ipsum dolor sit amet consetetur sadipscing}}

073C. \parbox[strut=rule, width=37%fw, valign=middle, border=0.5pt, minimize=true]{%
\test:fancymargins{\set[parameter=document.rskip, value=0]Lorem ipsum dolor sit amet consetetur sadipscing}}

\end{document}