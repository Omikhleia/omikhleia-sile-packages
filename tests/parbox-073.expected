Set paper size 	595.275597	841.8897729
Begin page
Mx 	29.7638
My 	54.0945
Set font 	Gentium Plus;10;400;;normal;;LTR
T	19 26 22 36 w=20.0293	(073A)
Mx 	49.7931
T	17 w=2.2900	(.)
Draw line	54.7757	46.2945	56.5342	0.5000
Draw line	54.7757	61.8945	56.5342	0.5000
Draw line	54.7757	46.2945	0.5000	16.1000
Draw line	111.3099	46.2945	0.5000	16.1000
Mx 	54.7757
My 	58.2945
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	83.8197
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	109.0198
T	17 w=2.2900	(.)
Mx 	29.7638
My 	76.6945
T	19 26 22 37 w=19.5752	(073B)
Mx 	49.3390
T	17 w=2.2900	(.)
Draw line	54.3101	62.8945	161.6016	0.5000
Draw line	54.3101	90.4945	161.6016	0.5000
Draw line	54.3101	62.8945	0.5000	28.1000
Draw line	215.9116	62.8945	0.5000	28.1000
Mx 	54.3101
My 	74.8945
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	83.3540
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	111.1968
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	135.7681
T	86 76 87 w=10.0146	(sit)
Mx 	148.4253
T	68 80 72 87 w=20.6836	(amet)
Mx 	171.7515
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	64.3101
My 	86.8945
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	29.7638
My 	105.2945
T	19 26 22 38 w=19.4385	(073C)
Mx 	49.2023
T	17 w=2.2900	(.)
Draw line	54.1694	91.4945	197.5059	0.5000
Draw line	54.1694	119.0945	197.5059	0.5000
Draw line	54.1694	91.4945	0.5000	28.1000
Draw line	251.6753	91.4945	0.5000	28.1000
Mx 	54.1694
My 	103.4945
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	83.2134
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	111.0561
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	135.6274
T	86 76 87 w=10.0146	(sit)
Mx 	148.2847
T	68 80 72 87 w=20.6836	(amet)
Mx 	171.6108
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	218.4136
T	86 68 71 76 83 w=21.6602	(sadip)
Mx 	240.0737
T	86 70 w=8.2324	(sc)
Mx 	248.3061
T	16 w=3.3691	(-)
Mx 	64.1694
My 	115.4945
T	76 81 74 w=13.1055	(ing)
End page
Finish
